.github/workflows/test.yml

```diff
@@ -3,14 +3,44 @@ on: pull_request
 env: { CLICOLOR_FORCE: 1 }
 
 jobs:
-  test:
+  cargo-fmt:
+    name: cargo-fmt@stable
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v3
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: rustfmt
       - name: Run formatting checks
         run: cargo fmt --check
+  cargo-clippy:
+    name: cargo-clippy@stable
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Rust stable
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          components: clippy
       - name: Run clippy checks
         run: cargo clippy --all-features -- -D warnings
+  cargo-test:
+    name: cargo-test@${{ matrix.version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [stable, beta, 1.63]
```

Comment:
Oh, one more thing: You should put quotes around the MSRV  (`"1.63"`). The reason is that otherwise it will get parsed as a number and only after that converted to a string.

That works as expected for `1.63`, but if you ever upgrade it to say `1.70` you'll get a surprise since the trailing zero gets dropped and you end up working with Rust 1.7.

Reply:
Oh alright, thanks a ton! Glad you caught that :)